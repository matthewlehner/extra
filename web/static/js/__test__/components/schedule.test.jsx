import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";

import Schedule from "components/schedule";

describe("Schedule component", () => {
  it("renders correctly", () => {
    const props = {
      toggleAutopilot: jest.fn()
    };
    const tree = renderer.create(
      <MemoryRouter>
        <Schedule {...props} />
      </MemoryRouter>
    );

    expect(tree).toMatchSnapshot();
  });
});
