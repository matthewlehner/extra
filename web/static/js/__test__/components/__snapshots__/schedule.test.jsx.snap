// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schedule component renders correctly 1`] = `
<section
  className="channel-schedule"
>
  <h2>
    Schedule
  </h2>
  <p>
    <span
      className="toggle-switch"
    >
      <input
        checked={true}
        className="input"
        id="toggler"
        name="checkbox"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="label"
        htmlFor="toggler"
      />
    </span>
    <strong>
      Schedule autopilot.
    </strong>
     Great for beginners and those with massive collections.
  </p>
  <div
    className="tab-container"
  >
    <nav
      className="tablist"
      role="tablist"
    >
      <a
        aria-controls="monday"
        aria-selected={true}
        className="tab"
        data-title="Monday"
        href="/?schedule=monday"
        id="monday-tab"
        onClick={[Function]}
        role="tab"
      >
        Monday
      </a>
      <a
        aria-controls="tuesday"
        aria-selected={false}
        className="tab"
        data-title="Tuesday"
        href="/?schedule=tuesday"
        id="tuesday-tab"
        onClick={[Function]}
        role="tab"
      >
        Tuesday
      </a>
      <a
        aria-controls="wednesday"
        aria-selected={false}
        className="tab"
        data-title="Wednesday"
        href="/?schedule=wednesday"
        id="wednesday-tab"
        onClick={[Function]}
        role="tab"
      >
        Wednesday
      </a>
      <a
        aria-controls="thursday"
        aria-selected={false}
        className="tab"
        data-title="Thursday"
        href="/?schedule=thursday"
        id="thursday-tab"
        onClick={[Function]}
        role="tab"
      >
        Thursday
      </a>
      <a
        aria-controls="friday"
        aria-selected={false}
        className="tab"
        data-title="Friday"
        href="/?schedule=friday"
        id="friday-tab"
        onClick={[Function]}
        role="tab"
      >
        Friday
      </a>
      <a
        aria-controls="saturday"
        aria-selected={false}
        className="tab"
        data-title="Saturday"
        href="/?schedule=saturday"
        id="saturday-tab"
        onClick={[Function]}
        role="tab"
      >
        Saturday
      </a>
      <a
        aria-controls="sunday"
        aria-selected={false}
        className="tab"
        data-title="Sunday"
        href="/?schedule=sunday"
        id="sunday-tab"
        onClick={[Function]}
        role="tab"
      >
        Sunday
      </a>
    </nav>
    <div
      aria-hidden={false}
      aria-labelledby="monday-tab"
      className="tabpanel"
      id="monday"
      role="tabpanel"
    >
      <table>
        <thead>
          <tr>
            <th>
              Time Slot
            </th>
            <th>
              Recurrence
            </th>
            <th>
              Collection
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <div
      aria-hidden={true}
      aria-labelledby="tuesday-tab"
      className="tabpanel"
      id="tuesday"
      role="tabpanel"
    >
      <table>
        <thead>
          <tr>
            <th>
              Time Slot
            </th>
            <th>
              Recurrence
            </th>
            <th>
              Collection
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <div
      aria-hidden={true}
      aria-labelledby="wednesday-tab"
      className="tabpanel"
      id="wednesday"
      role="tabpanel"
    >
      <table>
        <thead>
          <tr>
            <th>
              Time Slot
            </th>
            <th>
              Recurrence
            </th>
            <th>
              Collection
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <div
      aria-hidden={true}
      aria-labelledby="thursday-tab"
      className="tabpanel"
      id="thursday"
      role="tabpanel"
    >
      <table>
        <thead>
          <tr>
            <th>
              Time Slot
            </th>
            <th>
              Recurrence
            </th>
            <th>
              Collection
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <div
      aria-hidden={true}
      aria-labelledby="friday-tab"
      className="tabpanel"
      id="friday"
      role="tabpanel"
    >
      <table>
        <thead>
          <tr>
            <th>
              Time Slot
            </th>
            <th>
              Recurrence
            </th>
            <th>
              Collection
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <div
      aria-hidden={true}
      aria-labelledby="saturday-tab"
      className="tabpanel"
      id="saturday"
      role="tabpanel"
    >
      <table>
        <thead>
          <tr>
            <th>
              Time Slot
            </th>
            <th>
              Recurrence
            </th>
            <th>
              Collection
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
    <div
      aria-hidden={true}
      aria-labelledby="sunday-tab"
      className="tabpanel"
      id="sunday"
      role="tabpanel"
    >
      <table>
        <thead>
          <tr>
            <th>
              Time Slot
            </th>
            <th>
              Recurrence
            </th>
            <th>
              Collection
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</section>
`;
